prov/efa/src/rdm/efa_rdm_ep_utils.c

```diff
@@ -102,25 +89,25 @@ struct efa_rdm_peer *efa_rdm_ep_get_peer(struct efa_rdm_ep *ep, fi_addr_t addr)
 	if (peer)
 		return peer;
 
-	util_av_entry = ofi_bufpool_get_ibuf(ep->base_ep.util_ep.av->av_entry_pool,
-	                                     addr);
-	av_entry = (struct efa_av_entry *)util_av_entry->data;
+	EFA_INFO(FI_LOG_EP_DATA, "Creating peer for addr %lu\n", addr);
+	map_entry = ofi_buf_alloc(ep->peer_map_entry_pool);
+	if (OFI_UNLIKELY(!map_entry)) {
+		EFA_WARN(FI_LOG_EP_DATA,
+			"Map entries for fi_addr to peer mapping exhausted.\n");
+		return NULL;
+	}
 
-	if (av_entry->conn.ep_addr) {
-		av = ep->base_ep.av;
-		peer = (struct efa_rdm_peer *) ofi_buf_alloc(av->rdm_peer_pool);
-		if (!peer) {
-			EFA_WARN(FI_LOG_EP_DATA, "Failed to allocate peer struct\n");
-			return NULL;
-		}
+	map_entry->addr = addr;
 
-		efa_rdm_peer_construct(peer, ep, &av_entry->conn);
+	conn = efa_av_addr_to_conn(ep->base_ep.av, addr);
+	if (conn->ep_addr) {
+		efa_rdm_peer_construct(&map_entry->peer, ep, conn);
```

Comment:
Should you move the `map_entry` alloc inside this block?

Reply:
I removed the check altogether